scripts/cli/tasks/closeMilestone.ts

```diff
@@ -0,0 +1,93 @@
+import axios from 'axios';
+import _ from 'lodash';
+import { Task, TaskRunner } from './task';
+
+// const githubGrafanaUrl = 'https://github.com/grafana/grafana';
+
+interface CloseMilestoneOptions {
+  milestone: string;
+}
+
+const closeMilestoneTaskRunner: TaskRunner<CloseMilestoneOptions> = async ({ milestone }) => {
+  const gitHubUsername = process.env.GITHUB_USERNAME;
+  const gitHubToken = process.env.GITHUB_ACCESS_TOKEN;
+
+  if (!gitHubUsername || !gitHubToken) {
+    console.log(
+      'the close milestone operation needs a GITHUB_USERNAME and \
+    GITHUB_ACCESS_TOKEN environment variables'
+    );
+    return;
+  }
+
+  const client = axios.create({
+    baseURL: 'https://api.github.com/repos/grafana/grafana',
+    timeout: 10000,
+    auth: {
+      username: gitHubUsername,
+      password: gitHubToken,
+    },
+  });
+
+  const cherryPickLabel = 'cherry-pick needed';
+
+  if (!/^\d+$/.test(milestone)) {
+    console.log('Use milestone number not title, find number in milestone url');
+    return;
+  }
+
+  const res = await client.get(`/milestones/${milestone}`, {});
+
+  const milestoneData = res.data;
+
+  if (milestoneData.state === 'closed') {
+    console.log('milestone already closed. ✅');
+    return;
+  }
+
+  console.log('fetching issues/PRs of the milestone ⏬');
+
+  // Get all the issues/PRs with the label cherry-pick
+  // Every pull request is actually an issue
+  const issuesRes = await client.get('/issues', {
+    params: {
+      state: 'closed',
+      labels: cherryPickLabel,
+      per_page: 100,
+      milestone: milestone,
+    },
+  });
+
+  if (issuesRes.data.length < 1) {
+    console.log('no issues to remove label from');
+  } else {
+    console.log(`found ${issuesRes.data.length} issues to remove the cherry-pick label from 🔎`);
+  }
+
+  for (const issue of issuesRes.data) {
+    process.stdout.write(`🔧removing label from issue #${issue.number} 🗑...`);
```

Comment:
Use of `process.stdout.write` is due to wanting the `action -> result` logged in the same line.

Reply:
Would it make sense to put this comment into the code too? :)

Reply:
Absolutely!